
POLL_ERR and POLL_HUP are defined identically, looks like a typo,
The same value causes problems for "case POLL_ERR:", "case POLL_HUP:" handling.
It doesn't look like kernel generates these codes so far.

http://www.freebsd.org/cgi/query-pr.cgi?pr=126076


Index: sys/sys/signal.h
===================================================================
--- sys/sys/signal.h.orig
+++ sys/sys/signal.h
@@ -284,7 +284,7 @@
 #define POLL_MSG	3	/* Input message available		*/
 #define POLL_ERR	4	/* I/O Error				*/
 #define POLL_PRI	5	/* High priority input available	*/
-#define POLL_HUP	4	/* Device disconnected			*/
+#define POLL_HUP	6	/* Device disconnected			*/
 
 #endif
 
